### PR TITLE
Changed mmtable2() to gt() in Describe < Tables dialog

### LINKED ISSUE #8248: Moving from mmtable2() to gt()
@anastasia-mbithe after a discussion with @rdstern we have agreed that we should forgo the flexibility that `mmtable2` provides for `gt()` table directly. This is to allow for the flexibility that `gt()` offers.

Your work on the sub-dialogs is very impressive, and still holds really nicely so we don't need any changes on the sub dialogs from this move. In fact, doing this move will make things a lot easier on future sub-dialog code, and will add even more power to the tables!

The only changes are on the one variable, two/three variable, and (general) tables dialogs where we now remove all `mmtable2` code. 

E.g. we would replace
```
last_table <- mmtable2::mmtable(data=summary_table, cells=value) + mmtable2::header_left_top(variable='variable') + mmtable2::header_top_left(variable='summary')
```
with
```
last_table <- summary_table %>% gt()
```

I suggest we just focus on the "(general) tables" dialog for now, and can move to the other two later.

**Changes:**
There is one dialog change needed - and that is to remove the `Column Factors` nud and instead add in a set of rdos titled `Column Factor`.

If "Treat summaries as further factor" is **not** checked: Column Factor has three options - 
1. No column factor
2. Factor variable (if checked, a single receiver is displayed. This should be restricted to only variables in the factors receiver - this is implemented in another dialog, I'm just trying to remember an example. Do you know?)
3. Summary-Variable

If "Treat summaries as further factor" is checked: Column Factor has four options - the same two as above, but replace 3. with:
3. Summary
4. Variable

If 1. is checked, then the code is just:
`summary_table %>% gt()`

If 2. is checked, then the code is:
```
summary_table %>%
  pivot_wider(names_from = <variable in receiver>, values_from = value) %>%
  gt::gt()
```

If 3. is checked and "Treat summaries as further factor" is **not** checked, then the code is:
```
summary_table %>%
  pivot_wider(names_from = `summary-variable`, values_from = value) %>%
  gt::gt()
```

If 3 is checked and "Treat summaries as further factor" is checked, then the code is:
```
summary_table %>%
  pivot_wider(names_from = summary, values_from = value) %>%
  gt::gt()
```

If 4 is checked:
```
summary_table %>%
  pivot_wider(names_from = variable, values_from = value) %>%
  gt::gt()
```

Does this make sense?

### PR DESCRIPTION
**Pull request by anastasia-mbithe**
This fixes some of #8248 
@lilyclements and @rdstern This is ready for testing.

### CODE REVIEW

**Comment by lilyclements** [comment]
Great! To update, @anastasia-mbithe I will have a good review of this tomorrow and get back to you then. Do you have something to work on in the meantime?

@rdstern I suggest you hold off looking at this PR until after I've reviewed?

File: instat/dlgSummaryTables.vb [hunk 1]
```diff
@@ -21,18 +21,16 @@ Public Class dlgSummaryTables
...
 
     Private clsTableTitleFunction, clsTabFootnoteTitleFunction, clsTableSourcenoteFunction,
             clsCellTextFunction, clsCellBorderFunction, clsCellFillFunction, clsHeaderFormatFunction,
             clsTabOptionsFunction, clsBorderWeightPxFunction, clsFootnoteTitleLocationFunction, clsFootnoteSubtitleLocationFunction,
             clsTabFootnoteSubtitleFunction, clsStyleListFunction, clsFootnoteCellFunction, clsFootnoteCellBodyFunction,
             clsSecondFootnoteCellFunction, clsSecondFootnoteCellBodyFunction, clsTabStyleFunction, clsDummyFunction,
-            clsTabStyleCellTextFunction, clsTabStylePxFunction, clsTabStyleCellTitleFunction As New RFunction
+            clsTabStyleCellTextFunction, clsTabStylePxFunction, clsTabStyleCellTitleFunction, clsGtFunction As New RFunction
 
     Private clsMmtableOperator, clsSummaryOperator, clsFrequencyOperator, clsColumnOperator,
```
Comment (lilyclements) [inline instat/dlgSummaryTables.vb:35]: `clsMmtableOperator` can be removed (and any instances where you mention it)

File: instat/dlgSummaryTables.vb [hunk 2]
```diff
@@ -282,43 +273,19 @@ Public Class dlgSummaryTables
...
-        clsFrequencyOperator.SetOperation("+")
-        clsFrequencyOperator.AddParameter("mmtable2", clsRFunctionParameter:=clsMutableFunction, iPosition:=0)
+        clsFrequencyOperator.SetOperation("%>%")
+        clsFrequencyOperator.bBrackets = False
+        clsFrequencyOperator.AddParameter("tableFun", clsRFunctionParameter:=clsFrequencyDefaultFunction, iPosition:=0)
+        clsFrequencyOperator.AddParameter("gttbl", clsRFunctionParameter:=clsGtFunction, iPosition:=2)
 
         clsSummariesList.SetRCommand("c")
-        clsSummariesList.AddParameter("summary_mean", Chr(34) & "summary_mean" & Chr(34), bIncludeArgumentName:=False) ' TODO decide which default(s) to use?
+        clsSummariesList.AddParameter("summary_count", Chr(34) & "summary_count" & Chr(34), bIncludeArgumentName:=False) ' TODO decide which default(s) to use?
```
Comment (lilyclements) [inline instat/dlgSummaryTables.vb:288]: Out of interest, why is this changed?

**Review by lilyclements — CHANGES_REQUESTED** [review]
Really nice job @anastasia-mbithe! Looks great, and works really nicely.

A few minor comments below. And in addition to these:
- [ ] The receiver(s) clear on reopen (this occurs for factors and summaries)
- [ ] If I check the "Treat Summaries as Further Factor" checkbox, then select the "Variable" rdo, then uncheck the "Treat Summaries as Further Factor" checkbox, there is nothing selected for the rdo buttons. I suggest that if "Variable" or "Summary" is ever checked (and "Treat Summaries as Further Factor" is not checked), then "Summary-Variable" should be checked? (Otherwise, clicking and unclicking the checkbox shouldn't affect the rdo). What do you think?

File: instat/dlgSummaryTables.vb (hunk 2 above)
Comment (anastasia-mbithe) [reply instat/dlgSummaryTables.vb]: I guess it was some errors I was experiencing while coding, and count was working. but I have tested it and am not getting the error anymore, so I have changed it back to "summary_mean"

**Comment by anastasia-mbithe** [comment]
@lilyclements, the first comment was fixed in #8027, but I have fixed it here again to avoid the repetitive bug in case the other pull request takes time before being merged.
I have fixed the other comments as well. Please have a look.

**Comment by lloyddewit** [comment]
> the first comment was fixed in https://github.com/africanmathsinitiative/R-Instat/pull/8027, but I have fixed it here again to avoid the repetitive bug in case the other pull request takes time before being merged.

@rdstern I'm concerned that fixing the same issue in 2 different PRs may trigger a merge conflict. PR #8027 hopefully only needs a small regression test from you before we merge. It's been waiting for this test since 31 March so may have slipped under the radar. It would be great if we can merge PR #8027 fairly soon, thanks

**Review by rdstern — COMMENTED** [review]
@anastasia-mbithe and @lilyclements this seems to be working ok.  Great.  I can think of many improvements I would like, but would like to merge this version, (without (mmtable2) as soon as you feel it can be a replacement.  Then it become our new baseline.  The next priority will be the simpler job of replacing mmtable2 in the one-variable and 2/3 variable cases.  Then we return to this dialogue.

**Review by lilyclements — APPROVED** [review]
@anastasia-mbithe this is working great. A really nice job - well done. Happy for this to be merged 🥇 

@lloyddewit over to you!

**Review by rdstern — APPROVED** [review]

**Review by lloyddewit — APPROVED** [review]
@anastasia-mbithe clean, readable code as always - thanks

**Pull request merged**